src/main/scala/com/fulcrumgenomics/bam/FilterBam.scala

```diff
@@ -77,8 +87,12 @@ class FilterBam
     val kept = iterator.count { rec => {
         val throwOut = (removeDuplicates && rec.duplicate) ||
           (removeUnmappedReads && rec.unmapped) ||
+          (removeUnpairedReads && (!rec.paired || rec.unmapped || rec.mateUnmapped)) ||
```

Comment:
I guess it does remove fragment reads, as well as read pairs where both ends don't map.